index.js

```diff
@@ -110,43 +139,61 @@ class Assets {
     .then(resources => {
       // Process asset sets in parallel (up to 3)
       return BbPromise.map(assetSets, assets => {
-        // Try to resolve the bucket name
-        return this.resolveBucket(resources, assets.bucket)
-        .then(bucket => {
           const prefix = assets.prefix || '';
-          // Process files serially to not overload the network
-          return BbPromise.each(assets.files, opt => {
-            this.log(`Bucket: ${bucket}:${prefix}`);
-
-            if(this.options.bucket && this.options.bucket !== bucket) {
-              this.log('Skipping');
-              return;
+          // Try to resolve the bucket name
+          return this.resolveBucket(resources, assets.bucket)
+          .then((bucket) => {
+            if(assets.empty) {
+              this.log(`Emptying bucket`)
+              return this.emptyBucket(bucket, prefix)
+                .then(() => bucket)
             }
-
-            this.log(`Path: ${opt.source}`);
-
-            const cfg = Object.assign({}, globOpts, {cwd: opt.source});
-            glob.sync(opt.globs, cfg).forEach(filename => {
-
-              const body = fs.readFileSync(path.join(opt.source, filename));
-              const type = mime.lookup(filename) || opt.defaultContentType || 'application/octet-stream';
-
-              this.log(`\tFile:  ${filename} (${type})`);
-
-              const details = Object.assign({
-                ACL: assets.acl || 'public-read',
-                Body: body,
-                Bucket: bucket,
-                Key: path.join(prefix, filename),
-                ContentType: type
-              }, opt.headers || {});
-
-              return this.provider.request('S3', 'putObject', details);
-            });
-          });
-        });
-      }, { concurrency: 3 });
-    });
+            return Promise.resolve(bucket)
+          }).then(bucket => {
+            // Process files serially to not overload the network
+            return BbPromise.each(assets.files, (opt) => {
+              this.log(`Bucket: ${bucket}:${prefix}`)
+
+              if (this.options.bucket && this.options.bucket !== bucket) {
+                this.log('Skipping')
+                return
+              }
+
+              this.log(`Path: ${opt.source}`)
+
+              const cfg = Object.assign({}, globOpts, { cwd: opt.source })
+              glob.sync(opt.globs, cfg).forEach((filename) => {
+                const body = fs.readFileSync(path.join(opt.source, filename))
+                const type = mime.lookup(filename) || opt.defaultContentType || 'application/octet-stream'
+
+                this.log(`\tFile:  ${filename} (${type})`);
+
+                // when using windows path join resolves to backslashes, but s3 is expecting a slash
+                // therefore replace all backslashes with slashes
+                const key = path
+                  .join(prefix, filename)
+                  .split('\\')
```

Comment:
Why not just a replace, instead of split+join? 

`
.replace(/\\/g, '/')
`

Reply:
No particular reason. I've updated the PR to use `replace`